crates/cairo-coverage/src/input/mappings/statement_map.rs

```diff
@@ -0,0 +1,70 @@
+use crate::input::data_loader::deserialize;
+use crate::input::data_loader::types::{
+    CodeLocation, CoverageAnnotations, LineRange, ProfilerAnnotations,
+};
+use crate::types::{FileLocation, FunctionName};
+use anyhow::Result;
+use cairo_lang_sierra::debug_info::Annotations;
+use cairo_lang_sierra::program::StatementIdx;
+use derived_deref::Deref;
+use std::collections::HashMap;
+
+const PROFILER_NAMESPACE: &str = "github.com/software-mansion/cairo-profiler";
+const COVERAGE_NAMESPACE: &str = "github.com/software-mansion/cairo-coverage";
+
+#[derive(Deref)]
+pub struct StatementMap(HashMap<StatementIdx, StatementOrigin>);
```

Comment:
Well, the real reason we created the newtype here is that I wanted to use `From` to make it consistent with the architecture here, which in transition before there different states: input -> coverage_data -> output (currently lcov, but this will be a place for other formats if we wish for it). So, tl;dr, to use `From`.

Maybe we can work with naming to still use this as the newtype? Like suggested `StatementsMap` or `StatementsById`?

Reply:
Use functions instead traits imo, From trait does not bring much value here as it is a core bussines logic, not utility which will be reuse across the codebase.